crates/agglayer-config/src/prover.rs

```diff
@@ -67,21 +64,48 @@ impl Default for ProverConfig {
             max_concurrency_limit: default_max_concurrency_limit(),
             max_request_duration: default_max_request_duration(),
             max_buffered_queries: default_max_buffered_queries(),
-            cpu_prover: CpuProverConfig::default(),
-            network_prover: NetworkProverConfig::default(),
-            gpu_prover: GpuProverConfig::default(),
+            primary_prover: AgglayerProverType::NetworkProver(NetworkProverConfig::default()),
+            fallback_prover: None,
             grpc: Default::default(),
         }
     }
 }
 
+impl ProverConfig {
+    pub fn try_load(path: &Path) -> Result<Self, ConfigurationError> {
```

Comment:
Is it only for tests purpose? If so, can we flag it as is?

Reply:
At the moment it is for the test purpose. But it could be used in the code if needed.
Alternatively I should make a 
```
pub mod test {
     pub fn try_load_prover_config(path: &Path) -> Result<ProverConfig, ConfigurationError> {
}
```
as the `#[cfg(test)]` function could not be exported, even if used only in tests.
I think it would be better to stay as it is -  we have the same function for the `agglayer_config::Config`.

Reply:
If a function needs to be exported to other crate, while aiming at being use for tests, we have a dedicated `feature` that cover that in other crate under: `testutils`.

I'm fine letting this function public if you think it can be valuable soon.